integrationtest/backend/tck/src/main/java/org/hibernate/search/integrationtest/backend/tck/search/predicate/ObjectExistsSearchPredicateIT.java

```diff
@@ -115,6 +119,7 @@ public void nested() {
 				.fetchAllHits();
 
 		// DOCUMENT_2 won't be matched either, since it hasn't any not-null field
+		// DOCUMENT_4 won't be matched either, since we use a nested storage for the inner object field
```

Comment:
Even if I understand why it works that way, it's arguably more a bug than a feature. We should automatically add a join query ("nested" predicate) in that case. Maybe create a ticket related to the one that you created recently about implicit "nested" predicates, but specifically for the "exists" predicate? We may need specific handling.

Reply:
You're right. Done

Reply:
I couldn't find the ticket?

Reply:
Sorry I didn't do that. I did now: https://hibernate.atlassian.net/browse/HSEARCH-3762.